numpy/tests/typing/fail/scalars.py

```diff
@@ -32,11 +32,16 @@
 td_64 = np.timedelta64(1, "h")
 
 dt_64 + dt_64  # E: Unsupported operand types
-
 td_64 - dt_64  # E: Unsupported operand types
-td_64 / dt_64  # E: No overload
 td_64 % 1  # E: Unsupported operand types
-td_64 % dt_64  # E: Unsupported operand types
+
+# NOTE: The 2 tests below currently don't work due to the broad
+# (i.e. untyped) signature of `generic.__truediv__()` and `.__mod__()`.
+# TODO: Revisit this once annotations are added to the
+# `_ArrayOrScalarCommon` magic methods.
+
+# td_64 / dt_64  No overload
+# td_64 % dt_64  Unsupported operand types
```

Comment:
If you expect these to fail, I think this would be clearer as:

```suggestion
# td_64 / dt_64  # E: No overload
# td_64 % dt_64  # E: Unsupported operand types
```

Reply:
The `# E:` comment serves as a trigger for the `test_typing.py` script which is why I removed it.

Reply:
Hmm, shouldn't that script ignore lines with nothing on them?

Reply:
It would definitely be more convenient, but the current implementation simply searches for `"# E:"` sub-strings while iterating through an opened file.

https://github.com/numpy/numpy/blob/db866874ccb8c8c814df57eb1edca4fd05eb1753/numpy/tests/test_typing.py#L90-L101

Reply:
Perhaps that should become:
```python
code, marker = target_line.split("# E:")
if '#' not in code:
         marker = marker.strip() 
         assert lineno in errors, f'Extra error "{marker}"' 
         assert marker in errors[lineno] 
         continue
```

Reply:
Fixed in https://github.com/numpy/numpy/pull/17214/commits/b53bd5a06284bcd417e1b557e5515cbef9083b2c.